darts/tests/models/forecasting/test_torch_forecasting_model.py

```diff
@@ -1333,6 +1346,20 @@ def test_invalid_metrics(self):
             )
             model.fit(self.series)
 
+    def test_stateful_metrics(self):
+        torch_metrics = NumsCalled()
+        model = RNNModel(
+            12,
+            "RNN",
+            10,
+            10,
+            n_epochs=1,
+            torch_metrics=torch_metrics,
+            **tfm_kwargs,
+        )
+        model.fit(self.series)
+        assert model.model.trainer.logged_metrics["train_NumsCalled"] != 1
```

Comment:
Sorry, just saw this now. We should probably check that it is `> 1` since `0` would be incorrect.

```suggestion
        assert model.model.trainer.logged_metrics["train_NumsCalled"] > 1
```

Reply:
Good call. This tests that update_metrics works correctly as well